### ISSUE TITLE
backend error taking reference to `plus` in another module

### ISSUE THREAD

**Issue by jvasileff** [ceylon/ceylon-compiler#2378]
With:

```ceylon
native("jvm") module simple "1.0.0" {
  import ceylon.math "1.2.0";
}
```
```ceylon
import ceylon.math.whole {Whole}

shared void run() {
    value x = Whole.plus;
}
```

I get:

```
$ ceylon compile
source/simple/run.ceylon:4: warning: declaration is never used: 'x'
    value x = Whole.plus;
          ^
source/simple/run.ceylon:4: error: method plus in interface Summable<Other> cannot be applied to given types;
    value x = Whole.plus;
              ^
  required: Whole
  found: Summable
  reason: actual argument Summable cannot be converted to Whole by method invocation conversion
  where Other is a type-variable:
    Other extends Summable<Other> declared in interface Summable
ceylon compile: There was 1 error
```

I believe this is a recent regression.

**Comment by tombentley** [comment]
Yes, almost certainly a result of my fix for #2361.

**Comment by tombentley** [comment]
Should now be fixed.
